Added docs on auth setup (#95)
LykkeCorp/apollo: docs/Authentication/Clients.md

# Clients
Apollo requires two clients, one for the UI to authenticate against, and one for the API to use for updating claims.

## UI Client (`apollo_spa`)
`apollo_spa` requires the following settings to be configured. 

### RedirectUris
The following redirect uris are required for the ui client:

* http(s)://host/ui/auth/signinwin/main"
* http(s)://host/ui/auth/signinsilent/main"

### CORS Origins
The UI will need a CORS origin specifying:

* http(s)://host

**N.B.** There is no trailing slash

### Allowed Grant Types
Currently, the only grant type required is:

* Implicit

However, this will likely change as we move to a more secure model of authentication in the future.

### Scopes
The following scopes are required by the UI:

* openid
* profile
* [ConfiguredApi](./api_scopes.md)

## Apollo API Client (`apollo_api`)
`apollo_api` requires the following settings to be configured:

### Allowed Grant Types
The type of authentication for the API to talk to the Ironclad API is:

* ClientCredentials

> **N.B.** Using `ClientCredentials` requires a `ClientId` and `Secret` to be set fo rthe client

Reviewer: typo

Author: ```suggestion
**N.B.** Using `ClientCredentials` requires a `ClientId` and `Secret` to be set for the client
```

Author: Could've at least suggested the fix :)
